src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,6 +150,42 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
+     * Finds if inheritDoc is placed properly in java doc.
+     * @param ast Javadoc root node.
+     * @return true if inheritDoc is valid or false.
+     */
+    private static boolean isValidInheritDoc(DetailNode ast) {
+        final Set<Integer> allowedTypes = new HashSet<>();
+        allowedTypes.add(JavadocTokenTypes.NEWLINE);
+        allowedTypes.add(JavadocTokenTypes.LEADING_ASTERISK);
+        allowedTypes.add(JavadocTokenTypes.EOF);
+        boolean result = true;
+        boolean containsInheritDoc = false;
+        for (DetailNode child : ast.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.TEXT) {
+                if (!child.getText().trim().isEmpty()) {
+                    result = false;
```

Comment:
it is not a result, it is temporal state that non InheritDoc is found. So name variable appropriately - `nonInheritDocContent`, `extraTextFound`.
You need to finish execution asap if result of function is already known.